nuitka/nodes/BuiltinZipNodes.py

```diff
@@ -0,0 +1,242 @@
+                        source=ExpressionBuiltinIter1(
+                            value=ExpressionTempVariableRef(
+                                variable=zip_arg_variable, source_ref=self.source_ref
+                            ),
+                            source_ref=self.source_ref,
+                        ),
+                        source_ref=self.source_ref,
+                    )
+                ),
+                exception_name="TypeError",
+                handler_body=StatementRaiseException(
+                    exception_type=ExpressionBuiltinExceptionRef(
+                        exception_name="TypeError", source_ref=self.source_ref
+                    ),
+                    exception_value=makeConstantRefNode(
+                        constant="zip argument #%d must support iteration" % (i + 1),
+                        source_ref=self.source_ref,
+                    ),
+                    exception_trace=None,
+                    exception_cause=None,
+                    source_ref=self.source_ref,
+                ),
+                source_ref=self.source_ref,
+            )
+            for i, zip_iter_variable, zip_arg_variable in zip(
+                range(len(values)), zip_iter_variables, zip_arg_variables
+            )
+        ]
+
+        yield_statement = StatementExpressionOnly(
+            expression=ExpressionYield(
+                expression=ExpressionMakeTuple(
+                    elements=[
+                        ExpressionBuiltinNext1(
+                            value=ExpressionTempVariableRef(
+                                variable=zip_iter_variable, source_ref=self.source_ref
+                            ),
+                            source_ref=self.source_ref,
+                        )
+                        for zip_iter_variable in zip_iter_variables
+                    ],
+                    source_ref=self.source_ref,
+                ),
+                source_ref=self.source_ref,
+            ),
+            source_ref=self.source_ref,
+        )
+
+        loop_body = StatementLoop(
+            body=makeStatementsSequenceFromStatement(
+                statement=makeTryExceptSingleHandlerNode(
+                    tried=yield_statement,
+                    exception_name="StopIteration",
+                    handler_body=StatementLoopBreak(source_ref=self.source_ref),
+                    source_ref=self.source_ref,
+                )
+            ),
+            source_ref=self.source_ref,
+        )
+
+        statements.append(loop_body)
+
+        # Code generation expects this to be there.
+        statements.append(StatementGeneratorReturnNone(source_ref=self.source_ref))
+
+        generator_body.setBody(
+            makeStatementsSequence(
+                statements=statements, allow_none=False, source_ref=self.source_ref
+            )
+        )
+
+        return outline_body, "new_expression", "Lowered zip built-in to generator."
```

Comment:
This should return an iterator node, not the iterated value.

Reply:
`computeExpressionIter1` is expected to potentially change the iterator node iteration value, e.g. a list is lowered to a tuple, but returned an iterable value that is not iterated, is going to cause issues.